src/main/java/echoserver/ClientConnection.java

```diff
@@ -0,0 +1,40 @@
+package echoserver;
+
+import java.io.*;
+import java.net.Socket;
+
+public class ClientConnection {
+    public ClientConnection(){};
+
+    public ClientConnection(Socket socket) {
+        this.socket = socket;
+    }
+
+    public String promptInput() throws IOException {
+        lastSavedInput = convertInputStreamToString(socket.getInputStream());
+        return lastSavedInput;
+    }
+
+    public String readLastSavedInput(){
+        return lastSavedInput;
+    }
+
+    public OutputStream getOutputStream() throws IOException{
+        return socket.getOutputStream();
+    }
+
+    public Boolean isClosed(){
+       return socket.isClosed();
+    }
+
+    private
+
+    Socket socket;
+    String lastSavedInput;
+
+    String convertInputStreamToString(InputStream is) throws IOException{
+        byte[] bytes = new byte[1000];
```

Comment:
Why `1000`?

Reply:
There wasn't really a reason. I just chose a number and 1000 was the first that came to mind. I'm changing it to a `ByteArrayInputStream` object so it doesn't have a concrete capacity.